src/types/actions.js

```diff
@@ -128,7 +128,13 @@ type ReceiveProfileAction =
     }
   | { type: 'FATAL_ERROR_RECEIVING_PROFILE_FROM_STORE', error: Error }
   | { type: 'FATAL_ERROR_RECEIVING_PROFILE_FROM_URL', error: Error }
-  | {| +type: 'VIEW_PROFILE', +profile: Profile, +pathInZipFile?: string |}
+  | {|
+      +type: 'VIEW_PROFILE',
+      +profile: Profile,
+      +hiddenThreadIndexes: ThreadIndex[],
```

Comment:
The biggest change is that action has to decide the initial state of hidden threads based on the current state.